public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts

```diff
@@ -290,3 +274,20 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
     return this.templateSrv.getVariables().map((v) => `$${v.name}`);
   }
 }
+
+function migrateQuery(target: AzureMonitorQuery) {
+  if (target.queryType === AzureQueryType.ApplicationInsights) {
+    if ((target.appInsights as any).rawQuery) {
```

Comment:
This type cast seems error prone

Reply:
Yes 100% true. I actually have no idea what this all does (apart from 'migrate') - i just moved it out into a seperate function. I'll take a stab at it and see if i can clean it up.

Reply:
I tried to take a stab at improving this, but this seems wrong/I don't understand it, so I'm scared that making this typesafe will actually regress. 

I'm going to leave this as-is for now until I can get some guidance on what this code is supposed to actually do.